### ISSUE TITLE
Support `InputOptions.inject`

### ISSUE THREAD

**Issue by hyf0** [rolldown/rolldown#1578]
### What problem does this feature solve?

- https://github.com/rollup/plugins/tree/master/packages/inject#readme
- https://esbuild.github.io/api/#inject

### What does the proposed API look like?

.

**Comment by hyf0** [comment]
Block by oxc's solution on how identifying the same ast node between two visit.

**Comment by overlookmotel** [comment]
Instead of using `Span` to identify AST nodes, could use memory address of the `ImportDeclaration` as a unique ID.

```rs
fn get_stable_id(stmt: &ImportDeclaration) -> usize {
  stmt as *const _ as usize
}
```

NB: Needs to be address of the `ImportDeclaration` itself, NOT the `Statement` which wraps it. `Statement`s can move around in memory. Anything in a `Box<T>` will not move and therefore have a stable memory address.

NB2: Memory address will remain globally unique as long as you don't drop the allocator containing the AST.

The other way to identify statements is to use the index in the `Vec<Statement>`. This is preferable to pointers as then you don't need to search through all the statements for the span/pointer. You just index into `Vec<Statement>` - faster. But obviously the indexes will change if you insert new statements at top of the file.

**Comment by IWANABETHATGUY** [comment]
Another similar question I have found recently.

1. esbuild would generate Pure annotation attached to the newly generated node during `enum` transpiling. 

Since all **span** of newly generated ast node is `(0, 0)`, so we might also use **AST Node Id**  instead of `Span` to index leading comments to these **AST** nodes.

I hope to help you make better decisions by providing more use cases

**Comment by hyf0** [comment]
Sometimes I miss javascript, you could add almost anything on a object/struct and get it back :).

**Comment by underfin** [comment]
Related https://github.com/rolldown/rolldown/blob/7cef04fc78d021224a99bb9c1abb96964929a8fd/crates/rolldown_plugin_glob_import/src/lib.rs#L167-L167

**Comment by hyf0** [comment]
Closed by #1946.
